src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -38,7 +38,15 @@ def are_broadcast_compatible(left: TypeInfo, right: TypeInfo) -> bool:
     True
 
     """
-    if all([left.dims, right.dims]) and left.dims != right.dims:
+    both_dims_given = bool(left.dims and right.dims)
+    both_dims_given &= left.dims != GenericDimensions()
```

Comment:
If `GenericDimensions` is not a dataclass or defines a meaningful eq operator, most likely `left.dims != GenericDimensions()` would always be `True` because all instances of the class will compare as not equal.